docs/DeveloperGuide.md

```diff
@@ -1,34 +1,251 @@
+* You must have IntelliJ IDEA IDE installed on your computer.
+
+### 2.2 Setting up Fitr in your computer
+
+> :exclamation: **Caution:** Follow the steps in the following guide precisely. Things will not work out if you deviate in some steps.
+
+This section provides a step-by-step procedure to set up Fitr in your computer.
+
+1. **Fork** this [repository](https://github.com/AY2021S1-CS2113T-W13-2/tp/tree/v2.0).
+1. **Clone** the fork to your computer using [Sourcetree](https://sourcetreeapp.com/) or using any other _Git GUI_.
+1. Make sure your IDE is configured as **JDK 11**. If not, follow the steps below:
+    1. Ensure you have the correct JDK version installed in your computer.
+    1. Open IntelliJ (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+    1. Set up the correct JDK version for Gradle.
+        1. Click `Configure` > `Project Defaults` > `Project Structure`
+        1. Click `New...` and set it to the directory of the JDK.
+1. When prompted, **import** the project as a **Gradle project** (this can take a few minutes to complete). You can follow the following steps to import the project as a **Gradle Project**:
+
+    > :exclamation: **Note:** Importing a Gradle project is slightly different from importing a normal Java project.
+
+    1. IntelliJ IDEA by default has the Gradle plugin installed. If you have disabled it, click `File` > `Settings` > `Plugins` to enable them.
+    1. Click `Import Project` (or `Open or Import` in newer version of IntelliJ).
+    1. Locate the `build.gradle` file (not the root folder as you would do in normal importing) and select it. Click `OK`. If asked, choose to `Open as Project`, not `Open as File`.
+    1. Click `OK` to accept the default settings but do ensure that the selected version of `Gradle JVM` matches the JDK 11.
+
+### 2.3 Verifying Setup
+
+ 1. Run Fitr and enter a few commands to ensure that the application functions are as expected. You may refer to the _User Guide_ [here](https://ay2021s1-cs2113t-w13-2.github.io/tp/UserGuide.html) or enter `help` to find out what commands are supported in Fitr.
+ 1. Run the tests to ensure that they all pass, by executing the command `gradlew build` in IntelliJ's terminal.
+
+## 3. Design
+
+This section provides an overview of the design of the Fitr application.
+
+### 3.1 Architecture
+
+This section illustrates the high-level architecture of Fitr, as well as an illustration of the components used in Fitr.
+
+![Main class diagram](images/MainClassDiagram.png)
+
+_Figure 1: Architecture diagram of Fitr_
+
+Figure 1 above explains the high-level design of Fitr. Given below is a quick overview of each component.
+
+The main driver of the application is `Fitr` is responsible for mainly two phases:
+* At app launch: Initialises or loads the respective lists and user profile in the correct sequence and is in charge of connecting the various components with each other.
```

Comment:
As mentioned by Alicia, this part may be imprecise, and hence perhaps a adding a relevant diagram or a elaboration might help.